### PR TITLE
PTO module updated

### PR DESCRIPTION
**Pull request by Shruti-Apte**
Notion:
https://www.notion.so/saeloun/Update-the-designs-for-PTO-module-Admin-settings-084f3e014cb74b7a87b232e73ac732f9?pvs=4

##What:
- Separated Leaves and Holidays design.
- Updated code quality
- Fixed UI issues.

##Why
- Requirement 

Screenshot 2023-09-05 at 11 41 47 AM

Screenshot 2023-09-05 at 11 24 58 AM

Screenshot 2023-09-05 at 11 41 58 AM

Screenshot 2023-09-05 at 11 25 27 AM

### CODE REVIEW

File: app/javascript/src/components/Profile/Organization/Holidays/Details/index.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+              className="mt-2 rounded border border-miru-han-purple-1000 px-1 text-xs text-miru-han-purple-1000"
+              onClick={toggleCalendarModal}
+            >
+              View Calendar
+            </button>
+          </div>
+          <table>
+            <TableHeader />
+            <tbody>
+              {holidaysList.map((holiday, index) => (
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Profile/Organization/Holidays/Details/index.tsx:58]: what if `holidaysList` is empty []

File: app/javascript/src/components/Profile/Organization/Holidays/EditHolidays.tsx [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+}) => (
+  <div className="mt-4 p-4 lg:min-h-80v lg:bg-miru-gray-100 lg:p-10">
+    <div className="flex flex-col py-6 lg:flex-row">
+      <div className="flex p-2 lg:w-2/12">
+        <CalendarIcon className="mr-2 mt-1" size={20} weight="bold" />
+        <div className="">Public Holidays</div>
+      </div>
+      <div className="p-2 lg:w-10/12">
+        <div className="flex flex-col">
+          {holidayList.map((holiday, index) => (
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Profile/Organization/Holidays/EditHolidays.tsx:82]: what if `holidaysList` is empty []

File: app/javascript/src/components/Profile/Organization/Holidays/Details/index.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+              className="mt-2 rounded border border-miru-han-purple-1000 px-1 text-xs text-miru-han-purple-1000"
+              onClick={toggleCalendarModal}
+            >
+              View Calendar
+            </button>
+          </div>
+          <table>
+            <TableHeader />
+            <tbody>
+              {optionalHolidayList.map((holiday, index) => (
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Profile/Organization/Holidays/Details/index.tsx:81]: what if `optionalHolidayList` is empty []

File: app/javascript/src/components/Profile/Organization/Leaves/Details/index.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import TableRow from "./TableRow";
+
+const Details = ({ leavesList }) => (
+  <div className="mt-4 min-h-80v bg-miru-gray-100 p-4 lg:p-10">
+    <div className="flex w-full flex-col">
+      <div className="w-full pb-6 text-left text-xl font-semibold">Leaves</div>
+      <table className="flex w-full flex-col">
+        <TableHeader />
+        <tbody>
+          {leavesList.map((leave, index) => (
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Profile/Organization/Leaves/Details/index.tsx:13]: what if `leavesList` is empty []

File: app/javascript/src/components/Profile/Organization/Leaves/EditLeaves.tsx [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+    padding: 0,
+    cursor: "pointer",
+    background: state.isFocused || state.isSelected ? "#D7DEE5" : "#fff",
+  }),
+  "&:hover": {
+    background: "#D7DEE5",
+  },
+};
+
+const customStyles = {
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Profile/Organization/Leaves/EditLeaves.tsx:65]: Move `iconStyles `and `customStyles` to a separate file inside `EditLeaves`

File: app/javascript/src/components/Profile/Organization/Holidays/EditHolidays.tsx [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+import { Button } from "StyledComponents";
+
+import CustomDatePicker from "common/CustomDatePicker";
+import { CustomInputText } from "common/CustomInputText";
+import CustomReactSelect from "common/CustomReactSelect";
+import CustomToggle from "common/CustomToggle";
+import { Divider } from "common/Divider";
+import { repetitionType } from "constants/leaveType";
+
+const customStyles = {
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Profile/Organization/Holidays/EditHolidays.tsx:14]: Move `customStyles` to a separate file inside `EditHolidays`

File: app/javascript/src/components/Profile/Organization/Leaves/EditLeaves.tsx [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+  }),
+};
+
+const IconOption = props => (
+  <Option {...props} className="h-10 w-10 items-center">
+    <img alt={props.data.label} className="m-3 h-5 w-5" src={props.data.icon} />
+  </Option>
+);
+
+const ColorOption = props => (
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Profile/Organization/Leaves/EditLeaves.tsx:107]: Move` IconOption` and `ColorOption` to a separate file inside EditLeaves

File: app/javascript/src/components/Profile/RouteConfig.tsx [hunk 8]
```diff
@@ -9,8 +9,9 @@ import LeaveBalance from "./LeaveBalance";
 import Billing from "./Organization/Billing";
 import OrgDetails from "./Organization/Details";
 import OrgEdit from "./Organization/Edit";
+import Holidays from "./Organization/Holidays";
 import Import from "./Organization/Import";
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Profile/RouteConfig.tsx:13]: Please change the name of Import component to OrganizationImport

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

File: app/javascript/src/components/Profile/Organization/Holidays/Details/TableRow.tsx [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  const { date, name } = holiday;
+  const dateFormat = "DD.MM.YYYY";
+
+  return (
+    <tr className="flex w-full" key={key}>
+      <td
+        className="flex w-6/12 items-center border-b border-miru-gray-400 py-6 pr-2 text-left text-sm font-medium text-miru-dark-purple-1000 last:border-0"
+        scope="col"
+      >
+        {dayjs(date).format(dateFormat)}
```
[the remark this answers is not in the document]
Comment (Shruti-Apte) [reply app/javascript/src/components/Profile/Organization/Holidays/Details/TableRow.tsx:15]: No, need to fetch organisation date format from API. 
Currently using predefined date format.

**Review by prasanthchaduvula — APPROVED** [review]

**Pull request merged**